tests/unit/models/NotificationConfiguration.js

```diff
@@ -0,0 +1,216 @@
+const assert = require('assert');
+const { parseString } = require('xml2js');
+
+const NotificationConfiguration =
+    require('../../../lib/models/NotificationConfiguration.js');
+
+function checkError(parsedXml, err, errMessage, cb) {
+    const config = new NotificationConfiguration(parsedXml).
+        getValidatedNotificationConfiguration();
+    assert.strictEqual(config.error[err], true);
+    assert.strictEqual(config.error.description, errMessage);
+    cb();
+}
+
+function generateEvent(testParams) {
+    const event = [];
+    if (testParams.key === 'Event') {
+        if (Array.isArray(testParams.value)) {
+            testParams.value.forEach(v => {
+                event.push(`${event}<Event>${v}</Event>`);
+            });
```

Comment:
It might look better if you used map instead of forEach

```suggestion
            event.push(
                ...testParams.value.map(v => `${event}<Event>${v}</Event>`)
            );
```

Reply:
gonna leave the test file as is for the sake of time

Reply:
ok, will have to put it on the "come back" list.